### PR TITLE
1388 generate video caption

### PR DESCRIPTION
**Pull request by shidende**
This commit generate video augmentation using IBM Watson.

### CODE REVIEW

File: Assets/MirageXR/ContentTypes/Video/Editor/Mobile/CaptionGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+        #endregion
+
+        private int _recordingRoutine = 0;
+        private string _microphoneID = null;
+        private AudioClip _recording = null;
+        private int _recordingBufferSize = 1;
+        private int _recordingHZ = 22050;
+        private SpeechToTextService _service;
+        private string accumulatedText = "";
+        [SerializeField] private Button _stopButton;
```
Comment (cawke) [inline Assets/MirageXR/ContentTypes/Video/Editor/Mobile/CaptionGenerator.cs]: remove region comments and sort the fields: first private fields with the [SerializeField] attribute, then just private fields. write the attribute on the same line as the field.

Comment (cawke) [reply Assets/MirageXR/ContentTypes/Video/Editor/Mobile/CaptionGenerator.cs]: also follow the formatting throughout the file

File: Assets/MirageXR/ContentTypes/Video/Editor/Mobile/VideoEditorView.cs [hunk 2]
```diff
@@ -30,13 +33,36 @@ public class VideoEditorView : PopupEditorBase
...
+    [Space]
+    [SerializeField] private VideoPlayer videoPlayer;
+    [SerializeField] private Button playButton;
+    [SerializeField] private Button editCaption;
+    [SerializeField] private Button doneEditCaption;
+    [SerializeField] private GameObject _speechToText;
+    [SerializeField] private GameObject Toppanel;
+    [SerializeField] private GameObject Bottompanel;
+    [SerializeField] private GameObject VideoEditorPrefab;
+    [SerializeField] private GameObject VideoPreviePrefab;
```
Comment (cawke) [inline Assets/MirageXR/ContentTypes/Video/Editor/Mobile/VideoEditorView.cs]: if you write prefab in the field name, it means that you are going to store a link to the prefab that you are going to instantiate the object. For you, it’s just a link to a game object, which is confusing. rename the field

Comment (cawke) [reply Assets/MirageXR/ContentTypes/Video/Editor/Mobile/VideoEditorView.cs]: also follow the formatting and code style throughout the file

**Review by cawke — CHANGES_REQUESTED** [review]
image

There is no link to the script in the prefab.
because of this nothing works.

also write down a video explaining how it should work

**Comment by shidende** [comment]
Hi @cawke,
I have commited new changes to address your comments above. Also I have pushed the folder Demo that contains the video demostration.
Regards

**Comment by cawke** [comment]
> Hi @cawke, I have commited new changes to address your comments above. Also I have pushed the folder Demo that contains the video demostration. Regards

thanks, but delete the video from the project and upload it somewhere so we can watch it, for example here.

**Comment by shidende** [comment]
Here are the videos demo. 
https://github.com/WEKIT-ECS/MIRAGE-XR/assets/88036240/512ea7d0-4251-4bb4-a9fc-5b4673df8763
https://github.com/WEKIT-ECS/MIRAGE-XR/assets/88036240/18ffc2be-94fb-4eaa-8e74-f1c0a35eebcc
The video files already removed from the project.
CC: @cawke

**Comment by cawke** [comment]
> Here are the videos demo. https://github.com/WEKIT-ECS/MIRAGE-XR/assets/88036240/512ea7d0-4251-4bb4-a9fc-5b4673df8763 https://github.com/WEKIT-ECS/MIRAGE-XR/assets/88036240/18ffc2be-94fb-4eaa-8e74-f1c0a35eebcc The video files already removed from the project. CC: @cawke

ty. can you send me the key in slack so I can check the work? (and write where to put it)
image

File: Assets/MirageXR/ContentTypes/Video/Editor/Mobile/CaptionGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+            LoadKeysFromEnvFile();
+            Debug.Log("API Key: " + _iamApikey);
+            Debug.Log("URL: " + _serviceUrl);
+            LogSystem.InstallDefaultReactors();
+            Runnable.Run(CreateService());
+        }
+
+        private void LoadKeysFromEnvFile()
+        {
+            string pathToEnvFile = Application.dataPath + "/../ibm-credentials.env";
```
Comment (cawke) [inline Assets/MirageXR/ContentTypes/Video/Editor/Mobile/CaptionGenerator.cs]: file storage and downloading is not possible in this way.
it simply won’t make it into the build for mobile devices.
I'll rewrite this.

file storage and downloading is not possible in this way.
it simply won’t make it into the build for mobile devices.
I'll rewrite this.

the file with the keys will need to be renamed to "caption-ibm-credentials.txt" and placed in the path "{Project_Root}\Assets\Resources"

**Review by cawke — CHANGES_REQUESTED** [review]

**Review by cawke — APPROVED** [review]

**Pull request merged**